app/src/main/java/com/soen390/conumap/ui/directions/DirectionsFragment.kt

```diff
@@ -1,69 +1,61 @@
 package com.soen390.conumap.ui.directions
 
-import android.content.SharedPreferences
 import android.content.res.ColorStateList
-import android.graphics.Color
-import android.location.Address
-import android.location.Geocoder
-import androidx.lifecycle.ViewModelProviders
 import android.os.Bundle
-import android.util.Log
 import android.view.LayoutInflater
 import android.view.MotionEvent
 import android.view.View
 import android.view.ViewGroup
-import android.widget.TextView
-import android.widget.Toast
 import androidx.databinding.DataBindingUtil
 import androidx.fragment.app.Fragment
-import com.google.android.gms.maps.model.LatLng
 import androidx.fragment.app.activityViewModels
+import com.google.android.gms.maps.model.LatLng
 import androidx.navigation.fragment.NavHostFragment
 import com.soen390.conumap.R
 import com.soen390.conumap.databinding.DirectionsFragmentBinding
 import com.soen390.conumap.path.Path
-import com.soen390.conumap.ui.search_bar.SearchBarViewModel
 import kotlinx.android.synthetic.main.directions_fragment.*
-import java.util.*
 
 
 class DirectionsFragment : Fragment() {
-    var prefs: SharedPreferences? = null
-    val map = com.soen390.conumap.map.Map
-    var startLocationAddress:String?=null
-    var  startLocationText:String?=null
-    var endLocationText:String?=null
-    var startLocation:String?=null
-    var endLocation:String?=null
-    var endLocationAddress:String?=null
 
     companion object {
         fun newInstance() = DirectionsFragment()
     }
-
-    private lateinit var viewModel: DirectionsViewModel
     lateinit var binding : DirectionsFragmentBinding
-
+    val directionsViewModel: DirectionsViewModel by activityViewModels()
+    var path= Path
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
         var root = inflater.inflate(R.layout.directions_fragment, container, false)
-        val directionViewModel = ViewModelProviders.of(this)
-            .get(DirectionsViewModel::class.java)
-
 
         //This permit to inflate the fragment
         binding = DataBindingUtil.inflate<DirectionsFragmentBinding>(inflater, R.layout.directions_fragment, container, false).apply {
             this.setLifecycleOwner(activity)
-            this.viewmodel = directionViewModel
+            this.viewmodel=path
         }
 
-        getPath()
 
-        val sharedPreferences: SharedPreferences =requireContext().getSharedPreferences("SearchCount",0)
+        //Initial transportation mode is already driving
+        if (directionsViewModel.destinationChanged.value==null)
+        {
+            //If you're using an emulator, use the first commented out line
```

Comment:
Oh I just saw your comment here. I dont think we should hardcode the current location. Even on the emulator you can set your current location to whereever you want to.

Reply:
the thing is, the initial location for me is in California so I hard coded the location to somewhere closer to Montreal so I don't get memory overflow errors. 
I still kept the current location code to test. I can take it out and we can see with just that code